tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java

```diff
@@ -97,18 +109,93 @@ public ChunkWriterImpl(MeasurementSchema schema) {
     this.pageWriter = new PageWriter(measurementSchema);
     this.pageWriter.setTimeEncoder(measurementSchema.getTimeEncoder());
     this.pageWriter.setValueEncoder(measurementSchema.getValueEncoder());
+
+    //check if the measurement schema uses SDT
+    checkSdtEncoding();
+  }
+
+  public ChunkWriterImpl(MeasurementSchema schema, boolean isMerging) {
+    this(schema);
+    this.isMerging = isMerging;
+  }
+
+  private void checkSdtEncoding() {
+    if (measurementSchema.getProps() != null && !isMerging) {
+      for (Map.Entry<String, String> entry : measurementSchema.getProps().entrySet()) {
```

Comment:
No need to iterate the map, and also there is no guarantee that "loss" property will be got before other sdt parameters.

You can directly use map.contains and map.get() to get whatever property you want.

Reply:
will do. thank you. 
For "The paper calculate the FV of the next segment but you just use the endPoint of the last segment.", what do you mean by that? When ending segment0 and starting segment1, I use segment0's endpoint as segment1's startpoint. How do you calculate segment1's first value..?